src/extensionProvider.ts

```diff
@@ -93,65 +93,39 @@ export class ExtensionProvider implements IDappProvider {
     return !!this.account;
   }
 
-  async sendTransaction(transaction: ITransaction): Promise<ISignedTransaction> {
-    const txResponse = await this.startBgrMsgChannel("sendTransactions", {
-      from: this.account.address,
-      transactions: [transaction.toPlainObject()],
-    });
-
-    return this.getTransactionFactory().fromPlainObject(txResponse[0]);
+  async signTransaction<T extends ITransaction>(transaction: T): Promise<T> {
+    let response = await this.signTransactions([transaction]);
+    return response[0];
   }
 
-  async signTransaction(transaction: ITransaction): Promise<ISignedTransaction> {
-    const txResponse = await this.startBgrMsgChannel("signTransactions", {
+  async signTransactions<T extends ITransaction>(transactions: Array<T>): Promise<Array<T>> {
+    let extensionResponse = await this.startBgrMsgChannel("signTransactions", {
```

Comment:
this looks like a const, as it's not reassigned anywhere

Reply:
as regarding the startBgrMsgChannel(string), this looks error-prone, maybe we should move these to a constants file and use variables that have strings as values, it's much safer

Reply:
Fixed.

Fixed, added an enum.